lib/utils/patterns.ts

```diff
@@ -359,10 +359,31 @@ export function inferOrSelectDevice(preferredUuid: string) {
 	});
 }
 
+async function getApplicationByIdOrName(
+	sdk: BalenaSdk.BalenaSDK,
+	idOrName: string,
+) {
+	if (validation.looksLikeInteger(idOrName)) {
+		try {
+			return await sdk.models.application.get(Number(idOrName));
+		} catch (error) {
+			const { BalenaApplicationNotFound } = await import('balena-errors');
+			if (!(error instanceof BalenaApplicationNotFound)) {
+				throw error;
+			}
+		}
+	}
+	return await sdk.models.application.get(Number(idOrName));
```

Comment:
Looks like this should be just `idOrName` rather than `Number(idOrName)`.

Reply:
Good catch, fixed.